comment/mixins.py

```diff
@@ -1,162 +1,143 @@
+from comment.responses import UTF8JsonResponse
+from comment.models import BlockedUser
 
 
 class AJAXRequiredMixin:
+    status = 403
+    reason = ErrorMessage.NON_AJAX_REQUEST
+
     def dispatch(self, request, *args, **kwargs):
         if not request.META.get('HTTP_X_REQUESTED_WITH', None) == 'XMLHttpRequest':
-            return HttpResponseBadRequest(ErrorMessage.NON_AJAX_REQUEST)
+            data = {'status': self.status, 'reason': self.reason}
+            return UTF8JsonResponse(status=self.status, data=data)
         return super().dispatch(request, *args, **kwargs)
 
 
 class BasePermission(AJAXRequiredMixin):
-    def has_permission(self, request):
-        return True
+    reason = ErrorMessage.NOT_AUTHORIZED
 
-    def has_object_permission(self, request, obj):
+    def has_permission(self, request):
         return True
 
-
-class BaseCommentMixin(LoginRequiredMixin, BasePermission):
-    pass
-
-
-class BaseCommentView(FormView, DABResponseData):
-    form_class = CommentForm
-
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['comment_form'] = context.pop('form')
-        # context.update(get_comment_context_data(self.request))
-        context.update(DABContext(self.request))
-        return context
-
-    def get_form_kwargs(self):
-        kwargs = super().get_form_kwargs()
-        kwargs['request'] = self.request
-        return kwargs
+    def dispatch(self, request, *args, **kwargs):
+        if not self.has_permission(request):
+            data = {'status': self.status, 'reason': self.reason}
+            return UTF8JsonResponse(status=self.status, data=data)
+        return super().dispatch(request, *args, **kwargs)
 
 
-class CommentCreateMixin(BaseCommentView):
-    email_service = None
+class UserPermission(BasePermission):
+    reason = BlockedUserMSG.NOT_PERMITTED
 
-    def _initialize_email_service(self, comment, request):
-        self.email_service = DABEmailService(comment, request)
+    def has_permission(self, request):
+        return bool(
+            not settings.COMMENT_ALLOW_BLOCKING_USERS or
+            not BlockedUser.objects.is_user_blocked(request.user.id, request.POST.get('email'))
+        )
 
-    def _send_notification_to_followers(self, comment, request):
-        if settings.COMMENT_ALLOW_SUBSCRIPTION:
-            self._initialize_email_service(comment, request)
-            self.email_service.send_notification_to_followers()
 
-    def perform_save(self, comment, request):
-        comment.save()
-        self._send_notification_to_followers(comment, request)
-        comment.refresh_from_db()
-        return comment
+class BaseCommentPermission(UserPermission):
+    def has_permission(self, request):
+        if not request.user.is_authenticated:
+            self.reason = ErrorMessage.LOGIN_REQUIRED
+            return False
+        return super().has_permission(request)
```

Comment:
i think all the work done here is again done inside its child class `BaseCreatePermission`.  https://github.com/Radi85/Comment/pull/201/files#diff-a3711352663d6d29db96071e9efed0a4c0c9d556402c29d56111fcc47bda266dR59-R64

Is this really required?

Reply:
`BaseCreatePermission` has permission when anonymous system is enabled, all other acrions doesn't allow anonymity so they should be different.